panel/widgets/slider.py

```diff
@@ -182,16 +185,15 @@ def __init__(self, **params):
         if 'value' not in params:
             params['value'] = params.get('start', self.start)
         super().__init__(**params)
-
+        
     def _process_property_change(self, msg):
         msg = super()._process_property_change(msg)
-        if 'value' in msg:
-            msg['value'] = value_as_date(msg['value'])
-        if 'value_throttled' in msg:
-            msg['value_throttled'] = value_as_date(msg['value_throttled'])
+        if "value" in msg:
+            msg["value"] = datetime.strptime(msg['value'].strftime(self.format), self.format)
+        if "value_throttled" in msg:
+            msg["value_throttled"] = datetime.strptime(msg['value'].strftime(self.format), self.format)
```

Comment:
This logic is not correct. The `_process_property_change` method transforms the value sent from the frontend (usually an integer) into a datetime object. In the `_process_param_change` method you should do something like this.

Reply:
I put the functionality in `_process_property_change`  in my answer to @Hiescho, to always return the exact datetime(s) shown in the widget. This is to avoid something like this:
https://user-images.githubusercontent.com/19758978/141821143-20b23956-bcae-4ce6-9765-d0be2e8665d3.mp4

Is there a better way to do this? Or am I misunderstanding what is expected of the widget?

(The function should still use `value_as_date`)

Reply:
I am plotting monthly data and want to select start and end month of the range to be displayed with a DateRangeSlider. The DateRangeSlider does have a 'step' parameter, but it seems to be in units of milliseconds. So, as different months of the year have different lengths in time, it is not possible to make the slider handles move in steps of full calendar months by simply specifying a fixed step length. The next best solution I came up with is formatting the DateRangeSlider labels to display calendar months and year only, and discard the smaller timeframe parts of the slider values elsewhere in my code where necessary.

Reply:
So, it seems that _process_property_change is the right place to put the change in.